src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -146,6 +146,9 @@ public AsymmetricSignatureProvider(SecurityKey key, string algorithm, bool willC
             if (asymmetricAlgorithmResolver != null)
             {
                 AsymmetricAlgorithm asymmetricAlgorithm = asymmetricAlgorithmResolver(key, algorithm, willCreateSignatures);
+                if (asymmetricAlgorithm == null)
+                    throw LogHelper.LogException<ArgumentOutOfRangeException>(LogMessages.IDX10646, nameof(key), algorithm);
```

Comment:
@brentschmaltz  I am confused about the exception type that should be thrown here. `ArgumentOutOfRange` does not seem the right one. But I couldn't find any existing exception that can fit best.